test/functional/api/window_spec.lua

```diff
@@ -889,6 +889,253 @@ describe('API/win', function()
          }))
       eq(wins_before, meths.list_wins())
     end)
+
+    it('creates a split window', function()
+      local win = meths.open_win(0, true, {
+        win = 0,
+        vertical = false
+      })
+      eq(meths.win_get_config(win).relative, '')
+    end)
+
+    it('creates split windows in the correct direction', function()
+      local win = meths.open_win(0, true, {
+        vertical = true,
+        win = 0,
+      })
+      eq(meths.win_get_config(win).relative, '')
+
+      local layout = meths.call_function('winlayout', {})
+
+      -- walk the tree, and ensure the window is a vertical split
+      local queue = { layout }
+      local parent
+      local current
+      while true do
+        current = table.remove(queue)
+        if not current then break end
+        if current[1] == 'row' or current[1] == 'col' then
+          for _, child in ipairs(current[2]) do
+            if child[1] == "leaf" then
+              if child[2] == win.id then
+                parent = current
+                break
+              end
+            else
+              table.insert(queue, child)
+            end
+          end
+          if parent then break end
+        end
+      end
+      eq(parent[1], 'row')
+    end)
+
+    it('respects the \'split\' option', function ()
+      local win = meths.open_win(0, true, {
+        split = "below",
+        win = 0,
+      })
+      eq(meths.win_get_config(win).relative, '')
+
+      local layout = meths.call_function('winlayout', {})
+
+      -- walk the tree, and ensure the window is a horizontal split
+      local queue = { layout }
+      local parent
+      local current
+      while true do
+        current = table.remove(queue)
+        if not current then break end
+        if current[1] == 'row' or current[1] == 'col' then
+          for _, child in ipairs(current[2]) do
+            if child[1] == "leaf" then
```

Comment:
looks similar to the logic above, should it be extracted to a local function?

Reply:
Definitely could do, just wasn't sure if that was acceptable. Should I put that in the describe() block or in `helpers`?

I may also be able to get away with just asserting the result of `vim.fn.winlayout` instead

Reply:
> Should I put that in the describe() block

yes. helpers is for stuff that is useful for other test files.

> I may also be able to get away with just asserting the result of vim.fn.winlayout instead so maybe will get rid of this logic

even better :)